### PR TITLE
First pipeline

### PR DESCRIPTION
**Pull request by dalmouiee**
Pipeline for checking pushs to Main and pull requests into Main

### CODE REVIEW

**Comment by dalmouiee** [comment]
Interesting to see that the pipeline failed for python 3.6 and 3.9 but worked for 3.7 3.8. The fails were about cconflicting versions of SimpleITK that is defined for Platipy (when we import Platipy), and the SimpleITK version we define in the the requirements.txt.

**Comment by pchlap** [comment]
Yep that is interesting. I can make a change to platipy to clear up the issue for Python 3.9... The one for 3.6 might be a problem, because SimpleITK 2.0.0 requires 3.7 as a minimum. There was quite a few changes from SimpleITK 1 -> 2 so we need to talk about if we want to support both or not...

File: .github/workflows/python-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
+        pip install -r requirements.txt
+        pip install -r requirements-dev.txt
+    - name: Lint with flake8
```
Comment (pchlap) [inline .github/workflows/python-pipeline.yml:36]: Can we lint with pylint (instead of flake8)?

**Comment by dalmouiee** [comment]
We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?

**Review by pchlap — COMMENTED** [review]
> Yep that is interesting. I can make a change to platipy to clear up the issue for Python 3.9... The one for 3.6 might be a problem, because SimpleITK 2.0.0 requires 3.7 as a minimum. There was quite a few changes from SimpleITK 1 -> 2 so we need to talk about if we want to support both or not...

Actually, I think thats wrong. I think I just need to set them to the same version. Ill try to do that this afternoon.

**Comment by pchlap** [comment]
> We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?

Alternatively, can you try changing the required version of SimpleITK to 2.0.2 in this repo? I think that might fix it...

**Comment by dalmouiee** [comment]
> > We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?
> 
> Alternatively, can you try changing the required version of SimpleITK to 2.0.2 in this repo? I think that might fix it...

I changed this to 2.0.2, but now we get errors about pymedphys for python3.6 and platipy for python3.9, the other two are working fine

**Comment by pchlap** [comment]
> > > We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?
> > 
> > 
> > Alternatively, can you try changing the required version of SimpleITK to 2.0.2 in this repo? I think that might fix it...
> 
> I changed this to 2.0.2, but now we get errors about pymedphys for python3.6 and platipy for python3.9, the other two are working fine

Yeah pymedphys requires 3.7 as a minimum... I think we might need to support 3.7 as a minimum as well? At least for now?

**Comment by pchlap** [comment]
> > > > We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?
> > > 
> > > 
> > > Alternatively, can you try changing the required version of SimpleITK to 2.0.2 in this repo? I think that might fix it...
> > 
> > 
> > I changed this to 2.0.2, but now we get errors about pymedphys for python3.6 and platipy for python3.9, the other two are working fine
> 
> Yeah pymedphys requires 3.7 as a minimum... I think we might need to support 3.7 as a minimum as well? At least for now?

And to solve the issue in 3.9, try setting pydicom in pydicer to require at least 2.1.2... See if that helps. Not sure why this is only an error in Python 3.9 though...

**Comment by dalmouiee** [comment]
> > > > > We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?
> > > > 
> > > > 
> > > > Alternatively, can you try changing the required version of SimpleITK to 2.0.2 in this repo? I think that might fix it...
> > > 
> > > 
> > > I changed this to 2.0.2, but now we get errors about pymedphys for python3.6 and platipy for python3.9, the other two are working fine
> > 
> > 
> > Yeah pymedphys requires 3.7 as a minimum... I think we might need to support 3.7 as a minimum as well? At least for now?
> 
> And to solve the issue in 3.9, try setting pydicom in pydicer to require at least 2.1.2... See if that helps. Not sure why this is only an error in Python 3.9 though...

Yeah the issue is still persisting for 3.9, seems that there is some discrepancy between platipy and pydicom. Is it safe to assume it works for 3.7 and 3.8 for development now and leave out 3.9? It might cause us issues later on when we want to solve this issue when the tool is more complex... I'm running 3.8.10 on my mac but not sure about you and the others for development

**Comment by pchlap** [comment]
> > > > > > We could make the assumption that python3.7+ is in use for development and then sort this out later on for all version of Python 3?
> > > > > 
> > > > > 
> > > > > Alternatively, can you try changing the required version of SimpleITK to 2.0.2 in this repo? I think that might fix it...
> > > > 
> > > > 
> > > > I changed this to 2.0.2, but now we get errors about pymedphys for python3.6 and platipy for python3.9, the other two are working fine
> > > 
> > > 
> > > Yeah pymedphys requires 3.7 as a minimum... I think we might need to support 3.7 as a minimum as well? At least for now?
> > 
> > 
> > And to solve the issue in 3.9, try setting pydicom in pydicer to require at least 2.1.2... See if that helps. Not sure why this is only an error in Python 3.9 though...
> 
> Yeah the issue is still persisting for 3.9, seems that there is some discrepancy between platipy and pydicom. Is it safe to assume it works for 3.7 and 3.8 for development now and leave out 3.9? It might cause us issues later on when we want to solve this issue when the tool is more complex... I'm running 3.8.10 on my mac but not sure about you and the others for development

Yeah I'd say lets disable 3.9 from the pipeline for now. I'll add an issue to investigate whats going on with platipy and we'll make sure we resolve that for a future version.

**Comment by dalmouiee** [comment]
@pchlap Removed 3.6/3.9 from pipeline, if all looks good, happy to confirm and merge :)

**Review by pchlap — APPROVED** [review]
Looks great, thanks @dalmouiee!!

**Pull request merged**